### PR TITLE
fix: Support async webpack config

### PR DESCRIPTION
**Pull request by bradyhanna**
Allows the provided `webpackConfig` to return a promise/be an async function

```js
// playroom.config.js
module.exports = {
  // ...
  webpackConfig: async () => ({
    ...await getAsyncConfig(),
    ...otherConfig,
  })
};
```

Love the project btw 😄

### CODE REVIEW

**Comment by markdalgleish** [comment]
Thanks for the PR! Really curious what the use case is?

**Comment by bradyhanna** [comment]
> Really curious what the use case is

Our main webpack config is async to support a flow with a MFA token input

**Comment by markdalgleish** [comment]
Is this something webpack itself supports in webpack config files?

**Comment by markdalgleish** [comment]
So, I just figured out how to Google stuff. Looks like it can—I had no idea. Given this is a standard thing, this definitely makes sense to me now.

File: lib/makeWebpackConfig.js [hunk 1]
```diff
@@ -171,7 +171,7 @@ module.exports = async (playroomConfig, options) => {
   };
 
   const theirConfig = playroomConfig.webpackConfig
-    ? playroomConfig.webpackConfig()
+    ? await Promise.resolve(playroomConfig.webpackConfig())
```
Comment (markdalgleish) [inline lib/makeWebpackConfig.js:174]: Is the `Promise.resolve` call redundant here?

Comment (maraisr) [reply lib/makeWebpackConfig.js:174]: I think maybe it is. Afaik you cant await something that isnt thenable? Or at least I've got eslint rules that complain when you're using await on non-thenables.

Comment (bradyhanna) [reply lib/makeWebpackConfig.js:174]: It doesn't look like `webpack-merge` supports promises https://github.com/survivejs/webpack-merge/issues/81

**Pull request merged**